packages/fxa-auth-server/lib/log.js

```diff
@@ -17,22 +17,44 @@ const Sentry = require('@sentry/node');
 const ISSUER = config.get('domain') || '';
 const CLIENT_ID_TO_SERVICE_NAMES = config.get('oauth.clientIds') || {};
 
+// Keeps track of how many times a logger name has been used.
+const _registered = {};
+
 function Lug(options) {
+  const name = options.name || 'fxa-auth-server';
   EventEmitter.call(this);
-  this.name = options.name || 'fxa-auth-server';
+
+  // Ensure that names are unique. If two loggers are created with the same
+  // name, a double log scenario occurs.
+  if (_registered[name] === undefined) {
+    this.name = name;
+  } else {
+    let i = 1;
+    while (_registered[`${name}-${i}`] !== undefined) {
```

Comment:
nit: Why not keep the latest or next count in ``_registered[`${name}`]``?

Reply:
That'd be a nice touch 👍 .